examples/hif2a/fit_to_multiple_rbfes.py

```diff
@@ -349,40 +346,33 @@ def save_in_memory_callback(results, stage):
                 # TODO: replace with a function that knows what to report about each handle type
                 print(f'updated {int(np.sum(update_mask))} params by between {np.min(increment[update_mask]):.4f} and {np.max(increment[update_mask])}')
 
-        # Note: for certain kinds of method-validation tests, these labels could also be synthetic
         t1 = time()
         elapsed = t1 - t0
 
         print(f'completed forcefield-updating step {step} in {elapsed:.3f} s !')
 
-        # also save information about this step's parameter gradient and parameter update
-        # checkpoint results to npz (overwrite
-        flat_theta_traj.append(np.array(flat_theta))
-        flat_grad_traj.append(flat_loss_grad)
-        loss_traj.append(loss)
-
-        # TODO: update path
-        path_to_du_dls = 'du_dls_checkpoint.npz'
+        # save du_dls snapshot
+        path_to_du_dls = output_path.joinpath(f'du_dls_snapshot_{step}.npz')
         print(f'saving du_dl trajs to {path_to_du_dls}')
         du_dls_dict = dict() # keywords here must be strings
-        for key in results_traj:
-            du_dls_dict[_stringify_key(key)] = _results_to_arrays(results_traj[key])[1] # TODO: make this less gross
+        for stage in {'solvent', 'complex'}:
```

Comment:
nit: `for stage in results_this_step.keys()`

Reply:
Resolved in https://github.com/proteneer/timemachine/pull/367/commits/a24361543e280abcd444aa08f756fb96489d3557